07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.c

```diff
@@ -0,0 +1,342 @@
+#include "port.h"
+
+ InterruptHandler interruptHandlers[6][8] = {
+    /* PORTA */
+    {
+        NULL,  // PIN_0
+        NULL,  // PIN_1
+        NULL,  // PIN_2
+        NULL,  // PIN_3
+        NULL,  // PIN_4
+        NULL,  // PIN_5
+        NULL,  // PIN_6
+        NULL   // PIN_7
+    },
+    /* PORTB */
+    {
+        NULL,  // PIN_0
+        NULL,  // PIN_1
+        NULL,  // PIN_2
+        NULL,  // PIN_3
+        NULL,  // PIN_4
+        NULL,  // PIN_5
+        NULL,  // PIN_6
+        NULL   // PIN_7
+    },
+    /* PORTC */
+    {
+        NULL,  // PIN_0
+        NULL,  // PIN_1
+        NULL,  // PIN_2
+        NULL,  // PIN_3
+        NULL,  // PIN_4
+        NULL,  // PIN_5
+        NULL,  // PIN_6
+        NULL   // PIN_7
+    },
+    /* PORTD */
+    {
+        NULL,  // PIN_0
+        NULL,  // PIN_1
+        NULL,  // PIN_2
+        NULL,  // PIN_3
+        NULL,  // PIN_4
+        NULL,  // PIN_5
+        NULL,  // PIN_6
+        NULL   // PIN_7
+    },
+    /* PORTE */
+    {
+        NULL,  // PIN_0
+        NULL,  // PIN_1
+        NULL,  // PIN_2
+        NULL,  // PIN_3
+        NULL,  // PIN_4
+        NULL,  // PIN_5
+        NULL,  // PIN_6
+        NULL   // PIN_7
+    },
+    /* PORTF */
+    {
+        NULL,  // PIN_0
+        NULL,  // PIN_1
+        NULL,  // PIN_2
+        NULL,  // PIN_3
+        NULL,  // PIN_4
+    }
+};
+
+
+
+
+
+void Port_Init( Port_ConfigType* ConfigPtr)
+{
+	uint8_t u8_Pin_Counter=0;
+	for(u8_Pin_Counter=0;u8_Pin_Counter < PORT_PINS_NUMS;u8_Pin_Counter++)
+	{
+	Port_Num port_num =ConfigPtr[u8_Pin_Counter].PortNum;
+	Port_PinNum pin_num =ConfigPtr[u8_Pin_Counter].PortPinNum;
+		/* TODO:initiate the clock mention in DataSheet, p340 */
+SET_BIT(RCGCGPIO,port_num);		
+		if(PORT_PIN_DIRECTION_OUT==ConfigPtr[u8_Pin_Counter].PortPinDirection)
+		{
+		  SET_BIT(GPIO_DIR_REG(port_num),pin_num);
+		
+	  	if(PORT_PIN_LEVEL_HIGH==ConfigPtr[u8_Pin_Counter].PortPinLevelValue)
+		     {
+		     SET_BIT(GPIO_DATA_REG(port_num),pin_num);		    
+				 }
+			else if(PORT_PIN_LEVEL_LOW==ConfigPtr[u8_Pin_Counter].PortPinLevelValue)
+	       {
+         CLR_BIT(GPIO_DATA_REG(port_num),pin_num);
+		     }
+				 else
+				 {
+					 /*invalid*/
+				 }
+		switch(ConfigPtr[u8_Pin_Counter].PortPinOutputCurrent)
```

Comment:
Switch case should have only one break

Reply:
done now you can check again the new commit

Reply:
Great, good job.